doc/changelog_guidelines.md

```diff
@@ -0,0 +1,36 @@
+# CHANGELOG Guidelines
+
+The following guidelines should be followed when updating the CHANGELOG:
+- There should be an entry for every version, that includes the version number and release date.
+- Entries should be focused on communicating user-facing changes, considering that the main consumers of the CHANGELOG are the end users of SPIRE.
+- The types of changes should be grouped using the following categories:
+  - **Added**: New features that impact in the user experience.
+  - **Changed**: Changes in existing functionality.
+  - **Deprecated**: Features that will be removed in a future release.
+  - **Removed**: Features removed in this release.
+  - **Fixed**: Regular bug fixes. Should describe what the user would be experiencing if they were encountering the issue that is now fixed.
+  - **Security**: Security-related fixes.
+
+Categories that don't have an entry for the release are omitted in the CHANGELOG.
+
+The following is an example that includes all the categories:
+
+## [a.b.c] - YYYY-MM-DD
+
+### Added
+- Added AWS PCA configurable allowing operators to provide additional CA certificates for inclusion in the bundle (#1574)
+
+### Changed
+- Envoy SDS support is now always on (#1579)
```

Comment:
Will GitHub automagically turn these PR IDs into hyperlinks when this is part of the repo? Would be nice to have direct traceability back to the PRs in the CHANGELOG entries.

Reply:
GitHub will not convert to hyperlinks in the CHANGELOG.md file, but it will do it in the Releases page (https://github.com/spiffe/spire/releases) when rendering the vX.X.X annotated tags. I think that it's convenient to be able to just copy/paste the content of the CHANGELOG when creating the tags.